client/src/legacy/entwine/SubsitesTreeDropdownField.js

```diff
@@ -0,0 +1,33 @@
+/* global jQuery */
+jQuery.entwine('ss', ($) => {
+  /**
+   * Choose a subsite from which to select pages.
+   * Needs to clear tree dropdowns in case selection is changed.
+   */
+  $('select.subsitestreedropdownfield-chooser').entwine({
+    onchange() {
+      // TODO Data binding between two fields
+      const name = this.attr('name').replace('_SubsiteID', '');
+      const field = $(`#Form_EditForm_${name}`).first();
+      field.setValue(0);
+      field.refresh();
+      field.trigger('change');
+    }
+  });
+
+  /**
+   * Add selected subsite from separate dropdown to the request parameters
+   * before asking for the tree.
+   */
+  $('.TreeDropdownField.SubsitesTreeDropdownField').entwine({
+    getAttributes() {
+      const fieldName = this.attr('id').replace('Form_EditForm_', '');
+      const subsiteID = $(`#Form_EditForm_${fieldName}_SubsiteID option:selected`).val();
+
+      const attributes = this._super();
+      attributes.data.urlTree += `?${fieldName}_SubsiteID=${subsiteID}`;
+      attributes.data.cacheKey = `${attributes.data.cacheKey.substring(0, 19)}_${subsiteID}`;
```

Comment:
why are we getting a .substring(0, 19) ?

Reply:
Because Entwine 🤷‍♂ 
Existing code, not altered in this PR - just moved into a new folder structure.
https://github.com/silverstripe/silverstripe-subsites/blob/fd10b868ec5dfccd8bd0273c92a30892e2af48cb/javascript/SubsitesTreeDropdownField.js#L27